pkg/noderesourcetopology/cache/overreserve.go

```diff
@@ -265,6 +278,23 @@ func (ov *OverReserve) FlushNodes(logID string, nrts ...*topologyv1alpha2.NodeRe
 	}
 }
 
+func (ov *OverReserve) ShouldUpdateCacheNow(nodeName string, nrtCache *topologyv1alpha2.NodeResourceTopology) bool {
+	nrtActual, err := ov.nrtLister.Get(nodeName)
+	if err != nil || nrtActual == nil {
+		return false
+	}
+
+	if len(nrtActual.TopologyPolicies) == 0 || len(nrtCache.TopologyPolicies) == 0 {
```

Comment:
in addition to the comment already made, `TopologyPolicies` is deprecated: we should stop checking it (different PR of course) and check the top-level NRT attributes instead.